src/main/java/com/iexec/core/config/ObservabilityConfiguration.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023-2023 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.core.config;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Metrics;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.info.BuildProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class ObservabilityConfiguration {
```

Comment:
If you want, I can:

Add a getAppName in VersionService and add a postConstruct in the Controller to create the metric (by calling VersionService).

OR

Keep only the controller (delete the service which is a bit too much here)


But I liked the idea of managing this metric independently in a Config class because it can be replicated in all projects (we don't necessarily have the controller everywhere).

Reply:
The idea for me is to have the controller everywhere.
It will be more work but every app should have the version controller and the version metric endpoint in my opinion.